File: LLMgeneratedCode/llama-3.R
# Load necessary libraries
library(dplyr)
library(reshape2)
library(ggplot2)

# Load data
df_training <- read.csv("df_training.csv")
df_test <- read.csv("df_test.csv")
ano_training <- read.csv("ano_training.csv")
ano_test <- read.csv("ano_test.csv")

# Merge training data with metadata
df_training <- merge(df_training, ano_training, by = "Sample_ID")

# Merge test data with metadata
df_test <- merge(df_test, ano_test, by = "Sample_ID")

# Fit model using Lasso regression
library(lassod)
model <- lassod(ga ~ ., data = df_training, 
                method = "L1", 
                alpha = 0.5, # adjust this value to change the level of regularization
                nboots = 100) 

# Predict on test set
predictions <- predict(model, newdata = df_test)

# Calculate RMSE on test set
rmse <- sqrt(mean((df_test$ga - predictions)^2))
print(paste("RMSE on test set:", rmse))

# Create scatter plot of predicted vs actual gestational age values for test set
ggplot(df_test, aes(x = ga, y = predictions)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1, color = "black") +
  labs(title = "Predicted vs Actual Gestational Age Values",
       subtitle = paste("RMSE:", rmse),
       x = "Actual Gestational Age (weeks)",
       y = "Predicted Gestational Age (weeks)") +
  theme_classic()

File: LLMgeneratedCode/llama-3.py
# Import necessary libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

# Load data
df_training = pd.read_csv('df_training.csv')
df_test = pd.read_csv('df_test.csv')

ano_training = pd.read_csv('ano_training.csv', index_col='Sample_ID')
ano_test = pd.read_csv('ano_test.csv', index_col='Sample_ID')

# Merge data with metadata
df_training['gestational_age'] = df_training['Sample_ID'].map(ano_training['gestational_age'])
df_test['gestational_age'] = df_test['Sample_ID'].map(ano_test['gestational_age'])

# Split data into features (X) and target variable (y)
X_train = df_training.drop(['gestational_age'], axis=1)
y_train = df_training['gestational_age']
X_test = df_test.drop(['gestational_age'], axis=1)
y_test = df_test['gestational_age']

# Train a random forest regressor model
model = RandomForestRegressor()
model.fit(X_train, y_train)

# Make predictions on the test set
predictions = model.predict(X_test)

# Evaluate the model using RMSE
rmse = np.sqrt(mean_squared_error(y_test, predictions))
print(f"RMSE: {rmse}")

# Generate a scatter plot of predicted vs actual values for the test set
plt.figure(figsize=(10,6))
sns.scatterplot(x=predictions, y=y_test)
plt.axhline(y=0, color='r', linestyle='--', label='y = 0')
plt.legend()
plt.title("Predicted vs Actual Gestational Age")
plt.xlabel("Predicted Gestational Age")
plt.ylabel("Actual Gestational Age")
plt.savefig("./out2/llama-3.py.pdf", format="pdf")
# Plot RMSE in the scatter plot legend
plt.legend([f"RMSE: {rmse}"])
